caddyhttp/basicauth/setup.go

```diff
@@ -81,17 +82,54 @@ func basicAuthParse(c *caddy.Controller) ([]Rule, error) {
 			args = c.RemainingArgs()
 			switch len(args) {
 			case 0:
-				// Assume single argument is path resource
-				rule.Resources = append(rule.Resources, val)
+				if strings.Contains(val, "/") {
+					// Assume single argument is path resource
+					rule.Resources = append(rule.Resources, val)
+				} else {
+					return rules, c.Errf("expected ressource starting with '/', got %q", val)
+				}
 			case 1:
-				if val == "realm" {
+				switch val {
+				case "realm":
 					if rule.Realm == "" {
 						rule.Realm = strings.Replace(args[0], `"`, `\"`, -1)
 					} else {
 						return rules, c.Errf("\"realm\" subdirective can only be specified once")
 					}
-				} else {
-					return rules, c.Errf("expecting \"realm\", got \"%s\"", val)
+				case "allowed_cidr":
```

Comment:
Let's change this to `insecure_allowed_cidr` in keeping with the proxy directive's `insecure_skip_verify`. At least this way people will know that what they're doing is insecure.

Reply:
Even better: `insecure_skip_auth_cidr` to make it clear that this doesn't just allow those IPs to authenticate, it skips authentication for those IPs.

Reply:
This is still a really bad idea IMO.

Why not just bind to a different interface and let your firewall do the work? This seems like a solution being implemented in the wrong place.

Reply:
Don't merge it if you don't feel confortable about it, I won't be offended :)
This is only a proposal.

My use-case is to access mysite.com, if I'm at home, there's no security, if I'm at work I'll have to login. This is only for personal, non critical usage. If I'm the only one with this need, it should not be merged upstream.

This is however possible to configure nginx for this precise use-case:
```
location / {
    satisfy any;
    allow x.x.x.x;
    allow x.x.x.x;

    deny all;
    auth_basic "Login required";
    auth_basic_user_file /somefile/containing/the/users;

    autoindex on;
    root /some/directory/containing/files;
}
```

Reply:
@mholt @gregdel I'm also interested in this use-case, so that makes two of us.